src/components/Profile/Profile.js

```diff
@@ -0,0 +1,74 @@
+import React, { useState, useEffect } from 'react';
+
+import {
+  Container,
+  Grid,
+} from "@material-ui/core";
+
+import { get, post } from '../../requests';
+
+import SolutionCard from '../SolutionCard/SolutionCard';
+
+const Profile = ({update, setUpdate, userId, setUserId}) => {
+  const [userSolutions, setUserSolutions] = useState([]);
+  const [userName, setUserName] = useState('');
+
+  const removeSolution = id => {
+    setUserSolutions(userSolutions.filter((solution => solution.id !== id)));
+  };
+
+  const addUser = event => {
+    post('users/', {username: userName})
+      .then(response => {
+        setUserId(response.data.id);
+      });
+      event.preventDefault();
+    };
+
+  useEffect(() => {
+    if (userId && !update) {
```

Comment:
I suggest getting rid of `update` and refreshing this list every time component mounts. If it will affect the performance, will consider changing it.